app/models/work_packages/costs.rb

```diff
@@ -125,13 +125,13 @@ def reassign_cost_entries_before_destruction(work_packages, user, ids)
 
         false
       else
-        condition = "work_package_id = #{reassign_to.id}, project_id = #{reassign_to.project_id}"
-        ::WorkPackage.update_cost_entries(work_packages.map(&:id), condition)
+        condition = "entity_type='WorkPackage', entity_id = #{reassign_to.id}, project_id = #{reassign_to.project_id}"
+        ::WorkPackage.update_cost_entries(work_packages, condition)
```

Comment:
TI realise this wasn't your choice of naming, but `condition` is a poor choice - the parameter is named `action`.

Passing a Hash rather than raw SQL string would seem more elegant.

```suggestion
        condition = "entity_type='WorkPackage', entity_id = #{reassign_to.id}, project_id = #{reassign_to.project_id}"
        ::WorkPackage.update_cost_entries(work_packages, { entity_type: "WorkPackage", entity_id: reassign_to.id, project_id: reassign_to.project_id })
```

Reply:
I opted to just update the code to make it work, and not cause more changes by updating the style. If you feel this should be updated, we can create a code maintenance ticket to fix this.